### PR TITLE
fix(server): hyphenated pascal case regex (#839)

### LINKED ISSUE #839: PascalCaseHttpHeadersRule is to strict
and e.g. doesn't allow `ETag` header which is a valid header name.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Fixes #839

### CODE REVIEW

File: server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt [hunk 1]
```diff
@@ -58,6 +58,7 @@ class PatternUtilTest {
     fun checkIsHyphenatedPascalCase() {
         assertTrue(isHyphenatedPascalCase("Test-Case"))
         assertTrue(isHyphenatedPascalCase("X-Flow-Id"))
+        assertTrue(isHyphenatedPascalCase("ETag"))
```
Comment (roxspring) [inline server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt]: _Is_ this valid pascal-case? Definitions I see around only talk about capitalizing the first letter of each word. I know that Java classes use this approach to cope with acronyms and initialisms in class names but I was never clear whether this was part of pascal-case.

Had assumed that #839 was about adding "ETag" to a whitelist.

**Comment by ddufourboivin-zalando** [comment]
👍

File: server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt (hunk 1 above)
Comment (maxim-tschumak) [reply server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt]: As far as I know, the pascal-case is not a standard. That's why I've been following my "intuition" and proposed this change (accepting acronyms at the beginning of each hyphenated word). I think a less strict regex is better than having a whitelist mechanism.

Are you ok with this approach?

Comment (roxspring) [reply server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt]: No real objections, but have been thinking about the inconsistency of these patterns lately and was wondering what definitions we were working to. Questions to my mind:

1. Shouldn't NonHyphenatedPascalCase be allowed to have uppercase acronyms at the start of each word too?
2. What about the camelCase variations - should _they_ allow uppercase acronyms (yes, but only in 2nd+ word??)
3. Several of these case patterns block numbers entirely but several allow them; seems unlikely to me that this is right!

No reason to block this PR - I probably ought to raise this in a separate issue.

Comment (maxim-tschumak) [reply server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt]: Yes, let's discuss this in a separate issue

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt (hunk 1 above)
Comment (roxspring) [reply server/src/test/java/de/zalando/zally/util/PatternUtilTest.kt]: Raised #842

**Pull request merged**